diff --git a/seckit/src/Form/SecKitSettingsForm.php b/seckit/src/Form/SecKitSettingsForm.php
index f908077..a541ba1 100644
--- a/seckit/src/Form/SecKitSettingsForm.php
+++ b/seckit/src/Form/SecKitSettingsForm.php
@@ -189,7 +189,7 @@ class SecKitSettingsForm extends ConfigFormBase {
     // CSP default-src directive.
     $form['seckit_xss']['csp']['default-src'] = [
       '#type' => 'textfield',
-      '#maxlength' => 1024,
+      '#maxlength' => 4096,
       '#default_value' => $config->get('seckit_xss.csp.default-src'),
       '#title' => 'default-src',
       '#description' => $this->t("Specify security policy for all types of content, which are not specified further (frame-ancestors excepted). Default is 'self'."),
@@ -197,7 +197,7 @@ class SecKitSettingsForm extends ConfigFormBase {
     // CSP script-src directive.
     $form['seckit_xss']['csp']['script-src'] = [
       '#type' => 'textfield',
-      '#maxlength' => 1024,
+      '#maxlength' => 4096,
       '#default_value' => $config->get('seckit_xss.csp.script-src'),
       '#title' => 'script-src',
       '#description' => $this->t('Specify trustworthy sources for &lt;script&gt; elements.'),
@@ -205,7 +205,7 @@ class SecKitSettingsForm extends ConfigFormBase {
     // CSP object-src directive.
     $form['seckit_xss']['csp']['object-src'] = [
       '#type' => 'textfield',
-      '#maxlength' => 1024,
+      '#maxlength' => 4096,
       '#default_value' => $config->get('seckit_xss.csp.object-src'),
       '#title' => 'object-src',
       '#description' => $this->t('Specify trustworthy sources for &lt;object&gt;, &lt;embed&gt; and &lt;applet&gt; elements.'),
@@ -213,7 +213,7 @@ class SecKitSettingsForm extends ConfigFormBase {
     // CSP style-src directive.
     $form['seckit_xss']['csp']['style-src'] = [
       '#type' => 'textfield',
-      '#maxlength' => 1024,
+      '#maxlength' => 4096,
       '#default_value' => $config->get('seckit_xss.csp.style-src'),
       '#title' => 'style-src',
       '#description' => $this->t('Specify trustworthy sources for stylesheets. Note, that inline stylesheets and style attributes of HTML elements are allowed.'),
@@ -221,7 +221,7 @@ class SecKitSettingsForm extends ConfigFormBase {
     // CSP img-src directive.
     $form['seckit_xss']['csp']['img-src'] = [
       '#type' => 'textfield',
-      '#maxlength' => 1024,
+      '#maxlength' => 4096,
       '#default_value' => $config->get('seckit_xss.csp.img-src'),
       '#title' => 'img-src',
       '#description' => $this->t('Specify trustworthy sources for &lt;img&gt; elements.'),
@@ -229,7 +229,7 @@ class SecKitSettingsForm extends ConfigFormBase {
     // CSP media-src directive.
     $form['seckit_xss']['csp']['media-src'] = [
       '#type' => 'textfield',
-      '#maxlength' => 1024,
+      '#maxlength' => 4096,
       '#default_value' => $config->get('seckit_xss.csp.media-src'),
       '#title' => 'media-src',
       '#description' => $this->t('Specify trustworthy sources for &lt;audio&gt; and &lt;video&gt; elements.'),
@@ -237,7 +237,7 @@ class SecKitSettingsForm extends ConfigFormBase {
     // CSP frame-src directive.
     $form['seckit_xss']['csp']['frame-src'] = [
       '#type' => 'textfield',
-      '#maxlength' => 1024,
+      '#maxlength' => 4096,
       '#default_value' => $config->get('seckit_xss.csp.frame-src'),
       '#title' => 'frame-src',
       '#description' => $this->t('Specify trustworthy sources for &lt;iframe&gt; and &lt;frame&gt; elements. This directive is deprecated and will be replaced by child-src. It is recommended to use the both the frame-src and child-src directives until all browsers you support recognize the child-src directive.'),
@@ -245,7 +245,7 @@ class SecKitSettingsForm extends ConfigFormBase {
     // CSP frame-ancestors directive.
     $form['seckit_xss']['csp']['frame-ancestors'] = [
       '#type' => 'textfield',
-      '#maxlength' => 1024,
+      '#maxlength' => 4096,
       '#default_value' => $config->get('seckit_xss.csp.frame-ancestors'),
       '#title' => 'frame-ancestors',
       '#description' => $this->t("Specify trustworthy hosts which are allowed to embed this site's resources via &lt;iframe&gt;, &lt;frame&gt;, &lt;object&gt;, &lt;embed&gt; and &lt;applet&gt; elements."),
@@ -253,7 +253,7 @@ class SecKitSettingsForm extends ConfigFormBase {
     // CSP child-src directive.
     $form['seckit_xss']['csp']['child-src'] = [
       '#type' => 'textfield',
-      '#maxlength' => 1024,
+      '#maxlength' => 4096,
       '#default_value' => $config->get('seckit_xss.csp.child-src'),
       '#title' => 'child-src',
       '#description' => $this->t('Specify trustworthy sources for &lt;iframe&gt; and &lt;frame&gt; elements as well as for loading Workers.'),
@@ -261,7 +261,7 @@ class SecKitSettingsForm extends ConfigFormBase {
     // CSP font-src directive.
     $form['seckit_xss']['csp']['font-src'] = [
       '#type' => 'textfield',
-      '#maxlength' => 1024,
+      '#maxlength' => 4096,
       '#default_value' => $config->get('seckit_xss.csp.font-src'),
       '#title' => 'font-src',
       '#description' => $this->t('Specify trustworthy sources for @font-src CSS loads.'),
@@ -269,7 +269,7 @@ class SecKitSettingsForm extends ConfigFormBase {
     // CSP connect-src directive.
     $form['seckit_xss']['csp']['connect-src'] = [
       '#type' => 'textfield',
-      '#maxlength' => 1024,
+      '#maxlength' => 4096,
       '#default_value' => $config->get('seckit_xss.csp.connect-src'),
       '#title' => 'connect-src',
       '#description' => $this->t('Specify trustworthy sources for XMLHttpRequest, WebSocket and EventSource connections.'),
@@ -277,7 +277,7 @@ class SecKitSettingsForm extends ConfigFormBase {
     // CSP report-uri directive.
     $form['seckit_xss']['csp']['report-uri'] = [
       '#type' => 'textfield',
-      '#maxlength' => 1024,
+      '#maxlength' => 4096,
       '#default_value' => $config->get('seckit_xss.csp.report-uri'),
       '#title' => 'report-uri',
       '#description' => $this->t('Specify a URL (can be relative to the Drupal root, or absolute) to which user-agents will report CSP violations. Use the default value, unless you have set up an alternative handler for these reports. Note that if you specify a custom relative path, it should typically be accessible by all users (including anonymous). Defaults to <code>@report-url</code> which logs the report data.', ['@report-url' => SeckitInterface::CSP_REPORT_URL]),
@@ -285,7 +285,7 @@ class SecKitSettingsForm extends ConfigFormBase {
     // CSP policy-uri directive.
     $form['seckit_xss']['csp']['policy-uri'] = [
       '#type' => 'textfield',
-      '#maxlength' => 1024,
+      '#maxlength' => 4096,
       '#default_value' => $config->get('seckit_xss.csp.policy-uri'),
       '#title' => 'policy-uri',
       '#description' => $this->t("Specify a URL (relative to the Drupal root) for a file containing the (entire) policy. <strong>All other directives will be omitted</strong> by Security Kit, as <code>policy-uri</code> may only be defined in the <em>absence</em> of other policy definitions in the <code>X-Content-Security-Policy</code> HTTP header. The MIME type for this URI <strong>must</strong> be <code>text/x-content-security-policy</code>, otherwise user-agents will enforce the policy <code>allow 'none'</code>  instead."),
